Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -728,62 +728,65 @@ public class GraphTraverser: GraphTraversing {
         return references
     }
 
+    @Atomic var conditionCache = [GraphEdge: PlatformCondition.CombinationResult]()
```

Comment:
> different dependency paths lead to the same transitive dependency with different conditions.

Introducing a cache to solve the non-determinism makes me skeptical. If different paths lead to different conditions, shouldn't we traverse all the paths and merge the conditions? Also, won't this non-determinism exist across Tuist runs?

Reply:
➕ 1
I have either misunderstood the problem, or we should merge the results rather than getting the first one we hit

Reply:
removing the `visited` check fixed the bug.  Depending on which node was visited first, we would end up skipping it with subsequent runs of `combinedCondition` and end up with an incorrect result.  Always walking all paths was the solution.  

The cache was added because always walking all paths quickly becomes expensive.

Reply:
>  Always walking all paths was the solution.

This is the part that I missed. Thanks for clarifying it 🙏🏼